--- origsrc/xdm-1.1.6/configure.ac	2007-08-16 16:46:47.000000000 -0500
+++ src/xdm-1.1.6/configure.ac	2007-08-17 13:37:10.187500000 -0500
@@ -242,9 +242,13 @@
 
 XDM_LIBS="$XDM_LIBS $XDM_OS_LIBS"
 
-PKG_CHECK_EXISTS(xinerama, [
-	AC_DEFINE([USE_XINERAMA], 1, 
-		[Define to 1 to use XINERAMA in greeter & chooser])
+# Xinerama support
+AC_ARG_ENABLE(xinerama,
+		AC_HELP_STRING([--enable-xinerama],
+			[Use Xinerama in greeter and chooser]),
+    [USE_XINERAMA="$enableval"],
+	PKG_CHECK_EXISTS(xinerama, [USE_XINERAMA="yes"], [USE_XINERAMA="no"]))
+if test "x$USE_XINERAMA" = "xyes" ; then
 	PKG_CHECK_MODULES(XINERAMA, xinerama)
 	XDM_CFLAGS="$XDM_CFLAGS $XINERAMA_CFLAGS"
 	XDM_LIBS="$XDM_LIBS $XINERAMA_LIBS"
@@ -252,7 +256,10 @@
 	CHOOSER_LIBS="$CHOOSER_LIBS $XINERAMA_LIBS"
 	GREETER_CFLAGS="$GREETER_CFLAGS $XINERAMA_CFLAGS"
 	GREETER_LIBS="$GREETER_LIBS $XINERAMA_LIBS"
-])
+	AC_DEFINE([USE_XINERAMA], 1, 
+		[Define to 1 to use XINERAMA in greeter & chooser])
+fi
+AM_CONDITIONAL(USE_XINERAMA, test "x$USE_XINERAMA" = "xyes")
 
 # Xft text drawing for the greeter screen
 AC_ARG_WITH(xft, 
--- origsrc/xdm-1.1.6/greeter/Makefile.am	2006-07-14 15:01:18.000000000 -0500
+++ src/xdm-1.1.6/greeter/Makefile.am	2007-08-17 13:37:10.203125000 -0500
@@ -14,7 +14,7 @@
 
 AM_CFLAGS = $(XDMGREET_CFLAGS) -I$(top_srcdir)
 
-libXdmGreet_la_LDFLAGS = -module -avoid-version
+libXdmGreet_la_LDFLAGS = -module -avoid-version -no-undefined
 endif
 
 if LINT
